tests/testFocalPlane.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python
+#
+# LSST Data Management System
+# Copyright 2008-2013 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+
+import numpy
+import unittest
+
+import lsst.daf.base as dafBase
+import lsst.afw.image as afwImage
+from lsst.afw.cameraGeom.testUtils import DetectorWrapper
+import lsst.utils.tests
+import lsst.meas.base.tests
+from lsst.meas.base.tests import (AlgorithmTestCase, SingleFramePluginTransformSetupHelper)
+
+class FocalPlaneTestCase(AlgorithmTestCase):
+    def setUp(self):
+        self.center = lsst.afw.geom.Point2D(50.1, 49.8)
+        self.bbox = lsst.afw.geom.Box2I(lsst.afw.geom.Point2I(-20, -30),
+                                        lsst.afw.geom.Extent2I(140,160))
+        self.dataset = lsst.meas.base.tests.TestDataset(self.bbox)
+        self.dataset.addSource(100000.0, self.center)
+        self.dw = DetectorWrapper()
+
+    def tearDown(self):
+        del self.center
+        del self.bbox
+        del self.dataset
+        del self.dw
+
+    def testJacobianPlugin(self):
+        task = self.makeSingleFrameMeasurementTask("base_FocalPlane")
+        exposure, catalog = self.dataset.realize(10.0, task.schema)
+        exposure.setDetector(self.dw.detector)
+        task.run(exposure, catalog)
+        record = catalog[0]
+        self.assertFalse(record.get("base_FocalPlane_flag"))
+        pointKey = lsst.afw.table.Point2DKey(catalog.schema["base_FocalPlane"])
+        point = record.get(pointKey)
+        self.assertAlmostEqual(point.getX(), 1.012, 3)
+        self.assertAlmostEqual(point.getY(), 1.006, 3)
```

Comment:
These numbers are correct, but:

* Where they come from isn't transparent -- the casual reader of the code just has to take it on trust that `point.getX()` should be `1.012`.
* They're dependent on the default arguments to the `DetectorWrapper` constructor, which makes the test fragile to changes elsewhere in the codebase.

You could avoid both of these pitfalls by using something like:
```python
self.assertAlmostEqual(point.getX(), self.dw.pixelSize[0] * (0.5 + self.center[0]), 3)
self.assertAlmostEqual(point.getY(), self.dw.pixelSize[1] * (0.5 + self.center[1]), 3)
```